src/main/java/org/minions/devfund/yuri/WhackAMole.java

```diff
@@ -0,0 +1,231 @@
+   *         <code>false</code> if it is not.
+   */
+  public boolean place(final int posX, final int posY) {
+    boolean flag = false;
+    if (isEmptyCell(posX, posY)) {
+      grid[posX][posY] = MOLE;
+      flag = true;
+    }
+    return flag;
+  }
+
+  /**
+   * It is in charge to insert Moles to moleGrid.
+    */
+  public void insertMoles() {
+    int accumulator = 0;
+    while (accumulator < MOLES_AMOUNT) {
+      if (place(getRandomNumber(), getRandomNumber())) {
+        accumulator++;
+      }
+    }
+  }
+
+  /**
+   * It is charge to perform Whack in a specific position.
+   * @param posX coordinate X on moleGrid.
+   * @param posY coordinate Y on moleGrid.
+   */
+  public void whack(final int posX, final int posY) {
+    if (isMole(posX, posY)) {
+      increaseScore();
+      decreaseAttemptsLeft();
+      decreaseMolesLeft();
+    } else {
+      decreaseAttemptsLeft();
+    }
+    grid[posX][posY] = WACKED;
+  }
+
+  /**
+   * It is in charge to verify if the cell is empty.
+   * @param posX coordinate X on moleGrid.
+   * @param posY coordinate Y on moleGrid.
+   * @return <code>true</code> if it is an empty cell;
+   *         <code>false</code> if it is not.
+   */
+  public boolean isEmptyCell(final int posX, final int posY) {
+    return grid[posX][posY] == EMPTYCELL;
+  }
+
+  /**
+   * It is in charge to verify if the cell has a mole.
+   * @param posX coordinate X on moleGrid.
+   * @param posY coordinate Y on moleGrid.
+   * @return <code>true</code> if there is a mole ;
+   *         <code>false</code> if it is not.
+   */
+  public boolean isMole(final int posX, final int posY) {
+    return grid[posX][posY] == MOLE;
+  }
+
+  /**
+   * This method initialize the moleGrid with empty values.
+   */
+  public void fillGrid() {
+    for (int row = 0; row < grid.length; row++) {
+      for (int column = 0; column < grid[row].length; column++) {
+        grid[row][column] = EMPTYCELL;
+      }
+    }
+  }
+
+  /**
+   * Get a a random value in range from 0 to dimension value configured.
+   *
+   * @return the random value.
+   */
+  public int getRandomNumber() {
+    return new Random().nextInt(dimension);
+  }
+
+  /**
+   * Prints the Grid without showing where the moles are.
+   */
+  public void printGridToUser() {
+    for (int row = 0; row < grid.length; row++) {
+      for (int column = 0; column < grid[row].length; column++) {
+        if (grid[row][column] == MOLE) {
+          System.out.print(String.format("%c ", EMPTYCELL));
+        } else {
+          System.out.print(String.format("%c ", grid[row][column]));
+        }
+      }
+      System.out.print("\n");
+    }
+  }
+
+  /**
+   * It is in charge to get the moleGrid dimension.
+   * @return  moleGrid dimension.
+   */
+  public int getDimension() {
+    return dimension;
+  }
+
+  /**
+   * It ins charge to get the moleGrid.
+   * @return the moleGrid matrix.
+   */
+  public char[][] getGrid() {
+    char[][] gridGame = this.grid;
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Immediately return this expression instead of assigning it to the temporary variable "gridGame". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1488)